src/perform_step/sdirk_perform_step.jl

```diff
@@ -513,6 +515,151 @@ end
   f(integrator.fsallast,u,p,t)
 end
 
+@muladd function perform_step!(integrator, cache::SDIRK22ConstantCache, repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack a,α,β = cache.tab
+  nlsolver = cache.nlsolver
+  alg = unwrap_alg(integrator, true)
+
+  # precalculations
+  γ = a*dt
+  γdt = γ*dt
+  update_W!(integrator, cache, γdt, repeat_step)
+
+  # initial guess
+  zprev = dt*integrator.fsalfirst
+  nlsolver.z = zprev
+
+  # first stage
+  nlsolver.tmp = uprev + γdt*integrator.fsalfirst
+  z = nlsolve!(integrator, cache)
+  nlsolvefail(nlsolver) && return
+  uprev = α*nlsolver.tmp + β*z
+
+  # final stage
+  γ = dt
+  γdt = γ*dt
+  update_W!(integrator, cache, γdt, repeat_step)
+  nlsolver.tmp = uprev + γdt*integrator.fsalfirst
+  z = nlsolve!(integrator, cache)
+  nlsolvefail(nlsolver) && return
+  u = nlsolver.tmp
+
+  if integrator.opts.adaptive
+    if integrator.iter > 2
+      # local truncation error (LTE) bound by dt^3/12*max|y'''(t)|
```

Comment:
I'm not sure this LTE is correct, since that's the one for Trapezoid? What's the true LTE? That would change one of the constants in here, which isn't a big deal but is probably worth mentioning.